test/core/client_channel/service_config_test.cc

```diff
@@ -137,6 +137,11 @@ class ErrorParser : public ServiceConfig::Parser {
 };
 
 void VerifyRegexMatch(grpc_error* error, const std::regex& e) {
+  // Regexes don't work in gcc4.8 and below, so just return true in those cases.
+  // We use 5 instead of 4.8 because floats don't seem to work in macros.
+  #if defined(__GNUC__) && __GNUC__ <= 5
```

Comment:
GCC version x.y.z defines __GNUC__ to x, __GNUC_MINOR__ to y, and __GNUC_PATCHLEVEL__ to z.

Reply:
you could do something like -
if defined(__GNUC__) &&  ((__GNUC__ < 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__) <= 8))

Reply:
Done. Thanks!